### PR TITLE
O3-1508:Replace the current implementations of Patient Search in Service queues with the intended Advanced Search

### PR DESCRIPTION
**Pull request by jwnasambu**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
I added an ExtensionSlot to enable a user do advanced search on the service queues app.

## Screenshots

*None.*

https://user-images.githubusercontent.com/33891016/206525019-0580793b-3439-49d9-86f7-f0fc6ec4bf1c.mp4

## Related Issue
Required if applicable.
If present, please link any related issue here, e.g. "https://issues.openmrs.org/browse/123").
Don't forget to remove the *None.* above if you do fill this section.
-->

## Other

*None.*

### CODE REVIEW

**Review by CynthiaKamau — COMMENTED** [review]
Thank you @jwnasambu for this. When i select a patient, a list of appointments is supposed to be displayed, if there no appointments, the start visit form is launched. With the extension, on selecting a patient, im redirected to the patient chart.Could you look into this?

**Comment by jwnasambu** [comment]
@CynthiaKamau Sure!  thanks for the review.

**Comment by vasharma05** [comment]
Hi @CynthiaKamau !
Thank you for pointing the issue. I have guided @jwnasambu on how she can solve the pointed issues.

The only caveat in here would be that we won't have any `Back to search results` button in the scheduled visits or the visit form, since there's no way currently, that I can keep the search overlay open even after selecting a patient.

If you really need this, I can work on the patient search to get this implemented.
Do let me know your thoughts.
Thanks!

**Comment by jwnasambu** [comment]
@CynthiaKamau Kindly feel free to review my PR at your convinient time please!

File: packages/esm-outpatient-app/src/patient-search/visit-form/visit-form.component.tsx [hunk 1]
```diff
@@ -218,16 +218,7 @@ const StartVisitForm: React.FC<VisitFormProps> = ({ patientUuid, toggleSearchTyp
           </Row>
         )}
         <div className={styles.backButton}>
```
Comment (CynthiaKamau) [inline packages/esm-outpatient-app/src/patient-search/visit-form/visit-form.component.tsx:220]: When someone discards the form and tries to add a new queue entry, the visit form is still launched

File: packages/esm-outpatient-app/src/active-visits/active-visits-table.component.tsx [hunk 2]
```diff
@@ -519,7 +535,7 @@ function ActiveVisitsTable() {
           </Button>
         </Tile>
       </div>
-      {showOverlay && <PatientSearch view={view} closePanel={() => setShowOverlay(false)} />}
```
Comment (CynthiaKamau) [inline packages/esm-outpatient-app/src/active-visits/active-visits-table.component.tsx:522]: You should also update this to launch the search extension as it breaks
```
<p className={styles.content}>{t('noPatientsToDisplay', 'No patients to display')}</p>
 <Button
     kind="ghost"
     size="sm"
     renderIcon={(props) => <Add size={16} {...props} />}
      onClick={() => setShowOverlay(true)}>
      {t('addPatientToQueue', 'Add patient to queue')}
</Button>
```

**Review by CynthiaKamau — COMMENTED** [review]
Kindly address the few comments, great job so far !

File: packages/esm-outpatient-app/src/active-visits/active-visits-table.component.tsx (hunk 2 above)
Comment (jwnasambu) [reply packages/esm-outpatient-app/src/active-visits/active-visits-table.component.tsx:522]: @CynthiaKamau, @hadijahkyampeire Kindly I have fixed the proposed changes.

**Review by hadijahkyampeire — APPROVED** [review]
LGTM, thanks @jwnasambu

**Pull request merged**